### ISSUE TITLE
Parameters defined as integerType are compared by value not by type

### ISSUE THREAD

**Issue by jennymolske** [pact-foundation/pact-jvm#1804]
Hello,

I've got the following pact defined in our android app:

`@Pact(provider = COOKBOOK, consumer = CONSUMER_ANDROID)
    fun myCookBookPact(builder: PactDslWithProvider): RequestResponsePact {
        val requestHeader = hashMapOf(CkApiHeader.AUTHORIZATION to AUTH_TOKEN)
        return builder
            .given(...)
            .uponReceiving(...)
            .path(...)
            .method("GET")
            .headers(requestHeader)
            .willRespondWith()
            .status(200)
            .headers(JSON_RESPONSE_HEADER)
            .body(myResponseBody())
            .toPact()
    }

    private fun myResponseBody(): PactDslJsonBody = PactDslJsonBody()
        .stringType("id", "user-123")
        .stringType("name", "myCookbook")
        .integerType("difficulty", 4)
        .asBody()`

In the response body I've defined the 'difficulty' to be any integer. When uploading the pact and trying to verify with the provider state, I get the following error:

   Actual: {"id":"user-1","name":"Mein Kochbuch","difficulty":0}
       Diff
       --------------------------------------
       Key: - is expected 
            + is actual 
       Matching keys and values are not shown
        {
       -  "difficulty": 4
       +  "difficulty": 0
        }
       Description of differences
       --------------------------------------
       * Expected 4 but got 0 at $.difficulty

Why does this test fail? I've defined difficulty to be any integer, therefore the test shouldn't compare the exact values but only the type. stringType definition of id and name seems to be working.

Anyone an idea how to fix this?

Greetings, Jenny

Edit: The libs I used
pact-consumer-junit = { module = "au.com.dius.pact.consumer:junit5", version.ref = "pactConsumer" }
pact-consumer-kotlin = { module = "au.com.dius.pact.consumer:kotlin", version.ref = "pactConsumer" }
with
pactConsumer = "4.6.9"

**Comment by rholshausen** [comment]
Looks like that field is missing from the actual payload:
```
 {"id":"user-1","name":"Mein Kochbuch","type":0}
 ```

**Comment by jennymolske** [comment]
Oh, damn. My bad...I forgot to update the payload after I renamed the parameter 🙈 Sorry. I've updated the thread.

But in the meantime I've been able to figure out the problem by myself. Our server verification uses ruby standalone pact which uses v2 specification. In our app we use the latest library, which uses pact spec v4. 

When I change the test method annotation from this 
`@PactTestFor(providerName = COOKBOOK, port = PORT, pactMethod = "getCookbookRecipesPact")`

to this

`@PactTestFor(providerName = COOKBOOK, port = PORT, pactMethod = "getCookbookRecipesPact", pactVersion = PactSpecVersion.V2)`

the we are able to use the v2 matchers and everything works fine :)
